test/limits.jl

```diff
@@ -40,6 +40,9 @@ end
     frame = Frame(modexs[5]...)
     if VERSION < v"1.4.0-DEV.475"
         @test Aborted(frame, 1).at.file == Symbol("util.jl")
+    elseif isdefined(Base, :Experimental) &&
+           isdefined(Base.Experimental, Symbol("@force_compile"))
+        @test Aborted(frame, 1).at.file == Symbol("timing.jl")
```

Comment:
Ooh, that's unfortunate. I don't immediately understand why it's different from the old heuristic, but I haven't poked at it. Can you file a Julia issue? I regard line-number issues as bugs.

Reply:
Is this really bug? The line numbers appearing in lowered code seems just identical:
> v1.7
```julia
julia> ex = Main, Base.parse_input_line("""
           x = 1
           for i = 1:10
               x += 1
           end
           let y = 0
               z = 5
           end
           if 2 > 1
               println("Hello, world!")
           end
           @elapsed sum(rand(5))
           """; filename="fake.jl"); Meta.lower(Main, ex)
:($(Expr(:toplevel, :(#= fake.jl:1 =#), :(x = 1), :(#= fake.jl:2 =#), :(for i = 1:10
      #= fake.jl:3 =#
      x += 1
  end), :(#= fake.jl:5 =#), :(let y = 0
      #= fake.jl:6 =#
      z = 5
  end), :(#= fake.jl:8 =#), :(if 2 > 1
      #= fake.jl:9 =#
      println("Hello, world!")
  end), :(#= fake.jl:11 =#), :($(Expr(Symbol("hygienic-scope"), quote
    #= timing.jl:297 =#
    while false
        #= timing.jl:297 =#
    end
    #= timing.jl:298 =#
    local t0 = time_ns()
    #= timing.jl:299 =#
    $(Expr(:escape, :(sum(rand(5)))))
    #= timing.jl:300 =#
    (time_ns() - t0) / 1.0e9
end, Base))))))
```
> nightly
```julia
julia> ex = Base.parse_input_line("""
           x = 1
           for i = 1:10
               x += 1
           end
           let y = 0
               z = 5
           end
           if 2 > 1
               println("Hello, world!")
           end
           @elapsed sum(rand(5))
           """; filename="fake.jl"); Meta.lower(Main, ex)
:($(Expr(:toplevel, :(#= fake.jl:1 =#), :(x = 1), :(#= fake.jl:2 =#), :(for i = 1:10
      #= fake.jl:3 =#
      x += 1
      #= fake.jl:4 =#
  end), :(#= fake.jl:5 =#), :(let y = 0
      #= fake.jl:6 =#
      z = 5
  end), :(#= fake.jl:8 =#), :(if 2 > 1
      #= fake.jl:9 =#
      println("Hello, world!")
  end), :(#= fake.jl:11 =#), :($(Expr(Symbol("hygienic-scope"), quote
    #= timing.jl:356 =#
    $(Expr(Symbol("hygienic-scope"), :($(Expr(:meta, :force_compile))), Base.Experimental))
    #= timing.jl:357 =#
    local t0 = time_ns()
    #= timing.jl:358 =#
    $(Expr(:escape, :(sum(rand(5)))))
    #= timing.jl:359 =#
    (time_ns() - t0) / 1.0e9
end, Base))))))
```

The actual difference seems to be that v1.7 produces `NewvarNode` while nightly doesn't:
> v1.7
```julia
julia> modexs = collect(ExprSplitter(Main, ex)); frame = Frame(modexs[5]...); frame.framecode.src
CodeInfo(
    @ fake.jl:11 within `top-level scope`
1 ─     Core.NewvarNode(:(#310#t0))
   ┌ @ timing.jl:297 within `macro expansion`
2 ┄│     goto #4 if not false
3 ─│     goto #2
   └
   ┌ @ timing.jl:298 within `macro expansion`
4 ─│     _J1 = ($(QuoteNode(time_ns)))()
│  └
│  ┌ @ timing.jl:299 within `macro expansion`
│  │     rand(5)
│  │     sum(%J5)
│  └
│  ┌ @ timing.jl:300 within `macro expansion`
│  │     ($(QuoteNode(time_ns)))()
│  │     ($(QuoteNode(-)))(%J7, _J1)
│  │     ($(QuoteNode(/)))(%J8, 1.0e9)
│  └
└──     return %J9
)
```

> nightly
```julia
julia> modexs = collect(ExprSplitter(Main, ex)); frame = Frame(modexs[5]...); frame.framecode.src
CodeInfo(
    @ fake.jl:11 within `top-level scope`
   ┌ @ timing.jl:356 within `macro expansion`
1 ─│     $(Expr(:meta, :force_compile))
│  │ @ timing.jl:357 within `macro expansion`
│  │     _J1 = ($(QuoteNode(time_ns)))()
│  │ @ timing.jl:358 within `macro expansion`
│  │     rand(5)
│  │     sum(%J3)
│  │ @ timing.jl:359 within `macro expansion`
│  │     ($(QuoteNode(time_ns)))()
│  │     ($(QuoteNode(-)))(%J5, _J1)
│  │     ($(QuoteNode(/)))(%J6, 1.0e9)
│  └
└──     return %J7
)
```

So I'd say the first line number of that chunk was `fake.jl` sort of accidentally.

Reply:
xref #500